src/plan_bot_test.py

```diff
@@ -285,3 +287,46 @@ def test_process_post_wont_reply_to_warren_plan_bot(
 
     mock_build_response_text.assert_not_called()
     mock_reply.assert_not_called()
+
+def test_get_trigger_line():
+    assert (
+        plan_bot.get_trigger_line(
+            "i love liz warren\n!WarrenPlAnBot do you love Liz too?\ntell me the truth"
+        )
+        == "do you love Liz too?"
+    )
+
+    assert (
+        plan_bot.get_trigger_line(
+            "I like it here\n!WarrenPlAnBot do you?\n!WarrenPlAnBot do you really?"
+        )
+        == "do you really?"
+    )
+
+    assert plan_bot.get_trigger_line("!WarrenPlanBot, what's up?") == "what's up?"
+    assert plan_bot.get_trigger_line("!WarrenPlanBot,what's good?") == "what's good?"
+    assert (
+        plan_bot.get_trigger_line(
+            "/u/WarrenPlanBot hi\n!WarrenPlanBot,what's good?\n/u/WarrenPlanBot hi\nwarrenplanbot ho"
+        )
+        == "what's good?"
+    )
+
+    assert plan_bot.get_trigger_line("!WarrenPlanBot, --parent what's up?") == "--parent what's up?"
+    assert plan_bot.get_trigger_line("!WarrenPlanBot,--parent what's up?") == "--parent what's up?"
+
+def test_process_flags():
+    assert (
+        plan_bot.process_flags("what's up")
+        == ("what's up", {"parent": False})
+    )
+
+    assert (
+        plan_bot.process_flags("--parent what's up")
+        == ("what's up", {"parent": True})
+    )
+
+    assert (
+        plan_bot.process_flags("--parent, what's up")
```

Comment:
We had discussed making this `--tell-parent`. I'm still a bit on the fence. We could make them all synonyms as `--tell-parent` seems a little more self-explanatory but `--parent` seems maybe nicer for long term use once/if folks understand what it does. Perhaps that would just make things more confusing though.

No one has _ever_ used the `help`, so I don't think we'll be able to rely on that for users to learn what different commands mean :D 

If I was using this, personally, I'd probably use the `--parent` form as it's shorter, but we had settled on `--tell-parent` in the previous discussion / ticket

Reply:
I just read the issue description because I thought it had been updated to the latest spec.